### PR TITLE
Use `^1.4` for `composer/composer`

### PR DESCRIPTION
**Pull request by sandermangel**
Some modules require a more recent version of composer. To allow for this we use "^1.4" for the composer version

### Description
Using the "1.4 and above" will mean it's backwards compatible for current use of 1.4 and uses newer versions in installations where possible.
This will allow for installing modules that use newer versions

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=14013) <br/>All committers have signed the CLA.

**Review by miguelbalparda — APPROVED** [review]

**Comment by miguelbalparda** [comment]
Thanks for your contribution Sander! This is going to be reviewed soon, I'll let you know if we have questions!

**Comment by magento-engcom-team** [comment]
Hi @miguelbalparda, thank you for the review.
[ENGCOM-809](https://jira.corp.magento.com/browse/ENGCOM-809) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
@sandermangel thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Comment by sandermangel** [comment]
@magento-engcom-team  thanks guys, happy to see this going through!

**Pull request merged**
